### PR TITLE
feat: labels

### PR DESCRIPTION
**Pull request by Chiragmodi01**
### Feat. Labels _(create, delete, attach, remove, filter)_

- User can **add new labels** in labels page.
- User can **delete labels in labels** page.
- User can **attach a label** while creating a new note.
- User can **change or remove label** from any note.
- User can **filter notes by labels**.

<details><summary><h3>Show GIF Preview</h3></summary>

labels-preview

</details>

## [Live Preview](https://625b1d87377d0c0009c424f6--tranquil-choux-2fb728.netlify.app)

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -12,36 +13,23 @@
...
-.App-light .NoNotes_desc {
-  color: grey;
+.App-light .backgroundOptions , .LabelOptions {
+  color: #f9fbfc;
+  background-color: var(--mild-green);
 }
-.App-light .ham-menu_icon:hover {
-  background-color: #c6d9e25b;
+.App-light .LabelOptions {
+  /* box-shadow: 0 2px 5px 3px var(--light-green); */
```
Comment (devdeadviz) [inline src/App.css]: Remove commented codes.

File: src/comps/CreateLabels.jsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+                </span>
+              </div>
+            )
+          })
+        }
+      </div>}
+    </div>
+  )
+}
+
```
Comment (aurobindoGupta) [inline src/comps/CreateLabels.jsx:58]: leave empty line at end of code to remove this symbol

File: src/comps/BackgroundOptions.jsx [hunk 3]
```diff
@@ -38,15 +38,15 @@ function BackgroundOptions({color, showBgOptions, note, fromCreateNote }) {
...
+        <input checked={note.color === colorsObj.teal} type="radio" name='color' value={colorsObj.teal} onClick={(e) => changeBgColor(e)} className={color === 'teal' ? "bgOption active" : "bgOption teal"} title="teal"/>
+        <input checked={note.color === colorsObj.pink} type="radio" name='color' value={colorsObj.pink} onClick={(e) => changeBgColor(e)} className={color === 'pink' ? "bgOption active" : "bgOption pink"} title="pink"/>
+        <input checked={note.color === colorsObj.red} type="radio" name='color' value={colorsObj.red} onClick={(e) => changeBgColor(e)} className={color === 'red' ? "bgOption active" : "bgOption red"} title="red"/>
+        <input checked={note.color === colorsObj.brown} type="radio" name='color' value={colorsObj.brown} onClick={(e) => changeBgColor(e)} className={color === 'brown' ? "bgOption active" : "bgOption brown"} title="brown"/>
+        <input checked={note.color === colorsObj.grey} type="radio" name='color' value={colorsObj.grey} onClick={(e) => changeBgColor(e)} className={color === 'grey' ? "bgOption active" : "bgOption grey"} title="grey"/>
```
Comment (dusariudaykumar) [inline src/comps/BackgroundOptions.jsx:49]: one suggestion makes a component for the radio and map over it.  (your preference!)

File: src/helpers/services/editNoteService.js [hunk 4]
```diff
@@ -13,5 +13,5 @@ const editNoteService = async(notesDispatch, note) => {
         console.log(e.message)
```
Comment (aurobindoGupta) [inline src/helpers/services/editNoteService.js:13]: why twice

**Review by aurobindoGupta — APPROVED** [review]
nice work.
clear components.
keep going.

File: src/comps/CreateLabels.jsx [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        {
+          labelsList.length !== 0 && labelsList.map((item) => {
+            return (
+              <div className="label-item-wrapper" key={item}>
+                <span className="label-item-icons-container">
+                  <MdLabel title="Delete label" size='1.3em' className='label-item-icon label'/>
+                  <MdDelete title="Delete label" size='1.3em' className='label-item-icon trash' onClick={() => removeLabelHandler(item)} />
+                </span>
+                <span className="label-item-container-right">
+                  <h3 className="lbel-item-text">{item}</h3>
```
Comment (devdeadviz) [inline src/comps/CreateLabels.jsx]: I think you have a typo here, in the className check once.

**Review by devdeadviz — APPROVED** [review]
- Try to not destructure your props initially, instead of passing them as objects and destructuring at that place where you are accepting.
- Try to break your code into smaller chunks.

File: src/comps/CreateLabels.jsx [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  const {label, setLabel, labelsList, setLabelsList} = useNote();
+  
+  const addLabelHandler = (item) => {
+    if(label === ''){
+      toast.error('Label cannot be empty!')
+    } else if(labelsList.includes(label)){
+      toast.error(`${label} already exists!`);
+      setLabel('')
+    } else {
+      toast.error(`${label} added to labels!`);
```
Comment (dusariudaykumar) [inline src/comps/CreateLabels.jsx:17]: On adding a  new label the toast message should be a success

File: src/comps/CreateLabels.jsx (hunk 5 above)
Comment (Chiragmodi01) [reply src/comps/CreateLabels.jsx]: fixed now!

File: src/helpers/services/archiveNoteService.js [hunk 7]
```diff
@@ -9,7 +9,6 @@ const archiveNoteService = async(notesDispatch, note) => {
     const noteToSend = {note : note}
     try {
         const res = await axios.post(`/api/notes/archives/${note._id}` , noteToSend, {headers: Headers});
-        console.log(res)
         notesDispatch({ type: "ARCHIVE_NOTE", payload: {note: res.data.notes, archiveNote: res.data.archives}})
     } catch(e) {
         console.log(e.message)
```
Comment (dusariudaykumar) [inline src/helpers/services/archiveNoteService.js:14]: instead of consoling this, you can show the error message into view (toast message).

**Review by dusariudaykumar — APPROVED** [review]
The UI looks awesome.
 Everything looks fine

**Pull request merged**